webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java

```diff
@@ -1,35 +1,81 @@
 import com.box.l10n.mojito.entity.Repository;
+import com.box.l10n.mojito.service.thirdparty.smartling.SmartlingOptions;
+import com.box.l10n.mojito.service.thirdparty.smartling.SmartlingUtils;
+import com.box.l10n.mojito.service.thirdparty.smartling.SmartlingFile;
+import com.box.l10n.mojito.service.tm.search.SearchType;
+import com.box.l10n.mojito.service.tm.search.StatusFilter;
+import com.box.l10n.mojito.service.tm.search.TextUnitDTO;
+import com.box.l10n.mojito.service.tm.search.TextUnitSearcher;
+import com.box.l10n.mojito.service.tm.search.TextUnitSearcherParameters;
 import com.box.l10n.mojito.smartling.AssetPathAndTextUnitNameKeys;
 import com.box.l10n.mojito.smartling.SmartlingClient;
 import com.box.l10n.mojito.smartling.request.Binding;
 import com.box.l10n.mojito.smartling.request.Bindings;
 import com.box.l10n.mojito.smartling.response.ContextUpload;
 import com.box.l10n.mojito.smartling.response.File;
 import com.box.l10n.mojito.smartling.response.StringInfo;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.stereotype.Component;
 
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import static com.box.l10n.mojito.service.thirdparty.smartling.SmartlingUtils.isPluralFile;
+
 @ConditionalOnProperty(value = "l10n.ThirdPartyTMS.impl", havingValue = "ThirdPartyTMSSmartling")
 @Component
 public class ThirdPartyTMSSmartling implements ThirdPartyTMS {
 
     static Logger logger = LoggerFactory.getLogger(ThirdPartyTMSSmartling.class);
 
-    @Autowired
-    SmartlingClient smartlingClient;
+    /*
+    Devisible by 6 so that the plurals won't be broken up between files
+    We force pull all 6 forms regardless of language so that the files
+    contain the same keys and can be properly connected to their source
+    file in smartling.
+    */
+    private static final int DEFAULT_BATCH_SIZE = 5004;
+
+    private final SmartlingClient smartlingClient;
+    private final AssetPathAndTextUnitNameKeys assetPathAndTextUnitNameKeys;
+    private final TextUnitSearcher textUnitSearcher; // TODO - Add a cached version of textunitsearcher
+    private final Integer batchSize;
+
+    private List<SmartlingFile> lastPushResult;
 
     @Autowired
-    AssetPathAndTextUnitNameKeys assetPathAndTextUnitNameKeys;
+    public ThirdPartyTMSSmartling(@Autowired SmartlingClient smartlingClient,
```

Comment:
isn't `@Autowired` only needed on the ctor?

Reply:
Ah good catch, I forgot to remove the annotation from the constructor parameters.